contracts/FlashLoanV1Router02.sol

```diff
@@ -0,0 +1,200 @@
+        uint deadline
+    ) public virtual override ensure(deadline) returns (uint amount) {
+        address pool = FlashLoanV1Library.poolFor(factory, token);
+        IFlashLoanV1Pool(pool).transferFrom(msg.sender, pool, liquidity);
+        amount = IFlashLoanV1Pool(pool).burn(to);
+    }
+    function removeLiquidityETH(
+        uint liquidity,
+        address to,
+        uint deadline
+    ) public virtual override ensure(deadline) returns (uint amountETH) {
+        amountETH = removeLiquidity(WETH, liquidity, address(this), deadline);
+        IWETH(WETH).withdraw(amountETH);
+        TransferHelper.safeTransferETH(to, amountETH);
+    }
+    function removeLiquidityWithPermit(
+        address token,
+        uint liquidity,
+        address to,
+        uint deadline,
+        bool approveMax, uint8 v, bytes32 r, bytes32 s
+    ) external virtual override returns (uint amount) {
+        address pool = FlashLoanV1Library.poolFor(factory, token);
+        uint value = approveMax ? uint(-1) : liquidity;
+        IFlashLoanV1Pool(pool).permit(msg.sender, address(this), value, deadline, v, r, s);
+        amount = removeLiquidity(token, liquidity, to, deadline);
+    }
+    function removeLiquidityETHWithPermit(
+        uint liquidity,
+        address to,
+        uint deadline,
+        bool approveMax, uint8 v, bytes32 r, bytes32 s
+    ) external virtual override returns (uint amountETH) {
+        address pool = FlashLoanV1Library.poolFor(factory, WETH);
+        uint value = approveMax ? uint(-1) : liquidity;
+        IFlashLoanV1Pool(pool).permit(msg.sender, address(this), value, deadline, v, r, s);
+        amountETH = removeLiquidityETH(liquidity, to, deadline);
+    }
+
+    /**
+     * @dev Get the Deerfi Pool that will be used as the source of a loan.
+     * @param token The loan currency.
+     * @return The Deerfi V2 Pool that will be used as the source of the flash loan.
+     */
+    function getPoolAddress(address token) public view returns (address) {
+        return FlashLoanV1Library.poolFor(factory, token);
+    }
+
+    /**
+     * @dev From ERC-3156. The amount of currency available to be lended.
+     * @param token The loan currency.
+     * @return The amount of `token` that can be borrowed.
+     */
+    function maxFlashLoan(address token) external view override returns (uint256) {
+        address poolAddress = getPoolAddress(token);
+        if (poolAddress != address(0)) {
+            uint256 balance = IERC20(token).balanceOf(poolAddress);
+            if (balance > 0) return balance - 1;
+        }
+        return 0;
+    }
+
+    /**
+     * @dev From ERC-3156. The fee to be charged for a given loan.
+     * @param token The loan currency.
+     * @param amount The amount of tokens lent.
+     * @return The amount of `token` to be charged for the loan, on top of the returned principal.
+     */
+    function flashFee(address token, uint256 amount) public view override returns (uint256) {
```

Comment:
in `02` test, I expect you test and use this method, instead of still replying on `hardcoded fee` or `getFee`

Reply:
Done.